litert/vendors/qualcomm/core/builders/conv3d_op_builder.cc

```diff
@@ -0,0 +1,124 @@
+// Copyright (c) Qualcomm Innovation Center, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "litert/vendors/qualcomm/core/builders/conv3d_op_builder.h"
+
+#include <array>
+#include <cstddef>
+#include <cstdint>
+#include <variant>
+#include <vector>
+
+#include "litert/vendors/qualcomm/core/builders/op_builder.h"
+#include "litert/vendors/qualcomm/core/tensor_pool.h"
+#include "litert/vendors/qualcomm/core/utils/log.h"
+#include "litert/vendors/qualcomm/core/wrappers/op_wrapper.h"
+#include "litert/vendors/qualcomm/core/wrappers/quantize_params_wrapper.h"
+#include "litert/vendors/qualcomm/core/wrappers/tensor_wrapper.h"
+#include "QnnOpDef.h"  // from @qairt
+#include "QnnTypes.h"  // from @qairt
+
+namespace qnn {
+
+namespace {
+constexpr size_t kInputIndex = 0;
+constexpr size_t kFilterIndex = 1;
+constexpr size_t kBiasIndex = 2;
+constexpr size_t kNumInputsBias = 3;
+constexpr size_t kOutputIndex = 0;
+constexpr size_t kDepthIndex = 1;
+constexpr size_t kHeightIndex = 2;
+constexpr size_t kWidthIndex = 3;
+constexpr size_t kFilterDepthIndex = 0;
+constexpr size_t kFilterHeightIndex = 1;
+constexpr size_t kFilterWidthIndex = 2;
+
+}  // namespace
+
+std::vector<OpWrapper> BuildConv3dOp(
+    TensorPool& tensor_pool, const std::vector<TensorWrapperRef>& inputs,
+    const std::vector<TensorWrapperRef>& outputs, const std::uint32_t stride_d,
+    const std::uint32_t stride_h, const std::uint32_t stride_w,
+    const std::uint32_t dilation_d, const std::uint32_t dilation_h,
+    const std::uint32_t dilation_w, const PaddingType padding_type) {
+  std::vector<OpWrapper> res;
+
+  // conv
+  OpWrapper& conv_op = CreateOpWrapper(res, QNN_OP_CONV_3D);
+  TensorWrapper& input_tensor = inputs[kInputIndex];
+  TensorWrapper& filter_tensor = inputs[kFilterIndex];
+  conv_op.AddInputTensor(input_tensor);
+  conv_op.AddInputTensor(filter_tensor);
+  if (inputs.size() == kNumInputsBias) {
+    conv_op.AddInputTensor(inputs[kBiasIndex]);
+  }
+  conv_op.AddOutputTensor(outputs[kOutputIndex]);
+
+  // stride param
+  const std::array<std::uint32_t, 3> stride_data{stride_d, stride_h, stride_w};
+  const std::vector<std::uint32_t> stride_shape{3};
+  auto& stride_tensor = tensor_pool.CreateStaticTensor(
+      QNN_DATATYPE_UINT_32, QuantizeParamsWrapperVariant{}, stride_shape,
+      sizeof(decltype(stride_data)::value_type) * stride_data.size(),
```

Comment:
NIT: shorter version
```suggestion
      sizeof(stride_data[0]) * stride_data.size(),
```

Reply:
Thanks for the comment! Updated in the latest commit.